Team9Project/src/com/indragie/cmput301as1/ExpenseItemAddActivity.java

```diff
@@ -103,4 +180,168 @@ protected void onDone() {
 		setResult(RESULT_OK, getResultIntent());
 		finish();
 	}
+	
+	//================================================================================
+	// Receipt-Image Capture and Handling
+	//================================================================================
+	
+	/**
+	 * Sets up an external storage directory for receipt images and
+	 * creates a file to store a new one, then passes an intent
+	 * to a Camera application to take a photo.
+	 */
+	protected void takePhoto() {
+	
+		// create folder to store receipt images
+		String folder = Environment.getExternalStorageDirectory()
+									.getAbsolutePath() + "/ExpenseReceipts";
+		File receiptFolder = new File(folder);
+		if (!receiptFolder.exists()) {
+			receiptFolder.mkdir();
+		}
+		
+		// create image file Uri
+		String receiptFilePath = folder + "/"
+				+ String.valueOf(System.currentTimeMillis()) + ".jpg";
+		File receiptFile = new File(receiptFilePath);
+		receiptFileUri = Uri.fromFile(receiptFile);
+		
+		// create and dispatch picture intent 
+		Intent intent = new Intent(MediaStore.ACTION_IMAGE_CAPTURE);
+		intent.putExtra(MediaStore.EXTRA_OUTPUT, receiptFileUri);
+		
+		startActivityForResult(intent, CAPTURE_IMAGE_ACTIVITY_REQUEST_CODE);
+	}
+	
+
+	/*  Elements of scalingImage method borrowed from
+	 *  http://stackoverflow.com/questions/3331527/android-resize-a-large-bitmap-file-to-scaled-output-file
+	 *  last accessed: 03/15/15 3:29pm
+	 */ 
+	/**
+	 * Resizes a .jpeg file to be under the maximum size
+	 * @param imageFileUri The Uri of the image file to be resized
+	 */
+	public void scaleImage(Uri imageFileUri) {
+		try {
+		    final int IMAGE_MAX_SIZE = 65536; 
+		    InputStream fin = new FileInputStream(imageFileUri.getPath());
+
+		    // Decode image size
+		    BitmapFactory.Options options = new BitmapFactory.Options();
+		    options.inJustDecodeBounds = true;
+		    BitmapFactory.decodeStream(fin, null, options);
+		    fin.close();
+
+		    int scale = 1;
+		    while ((options.outWidth * options.outHeight) * (1 / Math.pow(scale, 2)) > 
+		          IMAGE_MAX_SIZE) {
+		       scale++;
+		    }
+		    
+		    if (scale > 1) {
+		    	Bitmap bmp = null;
```

Comment:
It would be better if you factored out the code that creates the scaled `Bitmap` object into a separate method so that you can write a unit test for it (to make sure the image is scaled as you expect)